### DISCUSSION TITLE
YAML folded multiline strings interpreted as multiple commands

### DISCUSSION THREAD

**Discussion by daandemeyer** [community/community#26105] [Actions]
<p>When using YAML folded multiline strings (using ‘&gt;’ instead of ‘|’), each line is still interpreted as a separate command instead of the newlines just being replaced by spaces.</p>
<p>See <a href="https://yaml-multiline.info/" rel="nofollow noopener">https://yaml-multiline.info/</a> for more information on YAML folded multiline strings.</p>

**Comment by ethomson** [comment]
<p>Thanks for reporting this, we’re taking a look.</p>

**Comment by chrispat** [comment]
<p>Could you give an example of where you are using the folding syntax?</p>

**Comment by daandemeyer** [comment]
<p>I’ve made a branch in reproc that changes one of my CMake invocations to use a folded multiline string on Ubuntu and macOS:</p>
<p><a href="https://github.com/DaanDeMeyer/reproc/runs/217390896" rel="nofollow noopener">https://github.com/DaanDeMeyer/reproc/runs/217390896</a></p>
<p>I want to use folding to avoid the platform specific command continuation character (’’ on Linux/macOS bash, ‘^’ on Windows cmd). Using folding, all newlines get converted to spaces which removes the need for the platform specific continuation character.</p>
<p>Unfortunately, we can see that all Ubuntu and macOS builds fail in the Configure step due to folded multiline strings being interpreted as multiple commands (at least that seems to be what’s going on).</p>

**Comment by chrispat** [comment]
<p>Currently the run step just takes the multi-line contents and writes that to a file that is then executed using the specified shell program.  The parser does not make any decisions based on the particular type of multi-line specifier in the YAML.  </p>
<p>Unfortunately I think we may have an issue where changing that behavior now will potentially cause workflow files that are currently working to break.</p>

**Comment by daandemeyer** [comment]
<p>I realize that a Beta is not supposed to break stuff but it would be rather convenient if this could still be changed while Github Actions is still in Beta.</p>
<p>Of course, your team has a way better view on how much stuff is already depending on this behavior than me so I understand if it’s not possible to change this anymore.</p>

**Comment by chrispat** [comment, marked as answer]
<p><a class="mention" href="/daandemeyer">@daandemeyer</a> could you try formatting your command like this and see if you still see the issue</p>
<pre><code>- run: &gt;-
    cmake
    -B build
    -S .
    -G Ninja
    -DCMAKE_BUILD_TYPE=Debug
    -DBUILD_SHARED_LIBS=ON
    -DREPROC++=ON
    -DREPROC_TEST=ON
    -DREPROC_EXAMPLES=ON
    -DREPROC_WARNINGS_AS_ERRORS=ON
    -DREPROC_TIDY=ON
    -DREPROC_SANITIZERS=ON
</code></pre>
<p>I think the extra level of indent is informing the YAML parser to add additional new line characters.</p>

**Comment by jclem** [comment]
<p>To elaborate on the syntax here a bit more, this is one of the odder parts of the YAML specification:</p>
<blockquote>
<p>[scalar content] can be written with the folded style (denoted by “&gt;”) where each line break is folded to a space unless it ends an empty or a more-indented line.</p>
</blockquote>
<blockquote>
<p><a href="https://yaml.org/spec/1.2/spec.html#id2760844" rel="nofollow noopener">https://yaml.org/spec/1.2/spec.html#id2760844</a></p>
</blockquote>
<p>It’s on yaml-multiline.info, too, but to be honest I’ve never actually read the paragraph on that page where it happens to mention this :joy:</p>

**Comment by daandemeyer** [comment]
<p>That fixes the issue. Thanks!</p>

**Comment by daandemeyer** [comment]
<p>I agree, this doesn’t exactly seem intuitive for a config language. Hopefully this thread helps others that encounter the same issue.</p>
